src/preprocessing/preprocess.py

```diff
@@ -47,27 +49,31 @@
 
 
 class WikiFile(object):
-    """Manage source file of a wiki page."""
+    """Manage the content of a wiki page."""
 
     def __init__(self, cwd, last3dirs, file_name):
         self.relative_path = join(last3dirs, file_name)
         self.url = file_name
         self._filename = join(cwd, file_name)
-        self._html = None
+        self._soup = None
 
-    def get_html(self):
-        """Return file content, load it if not loaded."""
-        if self._html is None:
+    @property
+    def soup(self):
+        """Return html soup of article, load content from file if needed."""
+        if self._soup is None:
             with open(self._filename, 'rb') as fh:
-                self._html = fh.read()
+                self._soup = bs4.BeautifulSoup(fh.read(), features='lxml', from_encoding='utf-8')
 
-        return self._html
+        return self._soup
 
-    def set_html(self, data):
-        """Set html source."""
-        self._html = data
+    @soup.setter
+    def set_soup(self, soup):
+        """Set html soup."""
+        self._soup = soup
 
-    html = property(get_html, set_html)
+    def get_html(self):
+        """Return current html as unicode string."""
+        return self.soup.decode()
```

Comment:
Trying to understand other thing, I've found that these .decode()/.encode() may take long. In my machine, the `.decode()` for a quite long article (the Argentina one) took 2 **seconds**.

So, I started to think what we could do to optimize this. the `get_html` function is only used to grab the page length. What do you think about removing this function completely, and storing here in Wikifile an attribute like `.original_html_length" that in `__init__` is set to None, but later is set to the real value when the HTML is read from disk?

Reply:
Buenísima la idea, corregido en b6f0c10.